eradiate/scenes/measure/_core.py

```diff
@@ -465,36 +502,78 @@ def to_dataset(self, aggregate_spps: bool = False) -> xr.Dataset:
         spectral_index = self._to_dataset_helper_spectral_index(spectral_coords)
 
         # Construct dataset
-        result = xr.Dataset(
-            data_vars=(
-                {
-                    "raw": (
-                        [spectral_coord_label, "sensor_id", "y", "x"],
-                        data,
-                        {"long_name": "raw sensor values"},
+        if eradiate.mode().has_flags(ModeFlags.ANY_RGB):
+            result = xr.Dataset(
+                data_vars=(
+                    {
+                        "raw": (
+                            [spectral_coord_label, "sensor_id", "y", "x", "channel"],
+                            data,
+                            {"long_name": "raw sensor values"},
+                        ),
+                        "spp": (
+                            [spectral_coord_label, "sensor_id"],
+                            spps,
+                            {"long_name": "sample count"},
+                        ),
+                    }
+                ),
+                coords={
+                    "x": ("x", xs, {"long_name": "film width coordinate"}),
+                    "y": ("y", ys, {"long_name": "film height coordinate"}),
+                    "channel": (
+                        "channel",
+                        ["R", "G", "B"],
+                        {"long_name": "color channel"},
                     ),
-                    "spp": (
-                        [spectral_coord_label, "sensor_id"],
-                        spps,
-                        {"long_name": "sample count"},
+                    spectral_coord_label: (
+                        spectral_coord_label,
+                        spectral_index,
+                        self._to_dataset_spectral_coord_metadata(),
```

Comment:
This dimension should be scalar: do you actually need it at all?

Reply:
Fair point (although I'm considering switching to well-specified dynamic dimensions, but that's to be left for the post-processing pipeline refactoring).